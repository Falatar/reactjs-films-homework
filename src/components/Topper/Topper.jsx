import React from 'react';
import style from './Topper.scss';
import Search from '../Search/Search';
import Film from '../Film/Film';
import Info from '../Info/Info';

function Topper() {
  const text = 'Adventure Drama Family Fantasy | 1h 46m';
  return (
    <div className={style.topper} id="space">
      <div className={style.searchLine}>
        <big>FILMS</big>
        <Search />
      </div>
      <div className={style.filmData}>
        <Film name="THE JUNGLE BOOK" genrePtime={text} rating={4.8} />
        <Info />
      </div>
    </div>
  );
}

export default Topper;
