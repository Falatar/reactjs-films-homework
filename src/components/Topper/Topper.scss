
div.topper {
    width: 100vw;
    height: 80vh;
    border: 0;
    padding: 0;
    margin: 0;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    background-image: url(../trash/jb_wall.jpg);
    background-size: cover;
    box-shadow: inset 0 110px 150px  -50px black , inset 0 -110px 150px  -50px black;
}

.topper {
    big {
        font-family: "Helvetica Neue", Helvetica, Arial, sans-serif;
        width: 10%;
        color: white;
        margin-left: 6%;
        margin-top: 1%;
        font-size: 1.5em;
        padding: 0;
        border: 0;
    }

    ::selection {
        background-color: dimgray;
    }
}


div.searchLine, div.filmData {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
}
