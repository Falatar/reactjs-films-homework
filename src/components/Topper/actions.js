const loadTopFilm = () => async (dispatch) => {
  const url = 'https://api.themoviedb.org/3/movie/top_rated?api_key=0f3cfa59da5e9c54b4eebea803330d71&language=en-US&page=1&adult=false';
  const promise = await fetch(url);
  const film = await promise.json();
  return dispatch({
    type: 'FORM_LATEST_FILM',
    payload: film,
  });
};

export const loadGenres = () => async (dispatch) => {
  const url = 'https://api.themoviedb.org/3/genre/movie/list?api_key=0f3cfa59da5e9c54b4eebea803330d71&language=en-US';
  const promise = await fetch(url);
  const list = await promise.json();
  return dispatch({
    type: 'LOAD_GENRES_TO_TOP',
    payload: list,
  });
};

export default loadTopFilm;
