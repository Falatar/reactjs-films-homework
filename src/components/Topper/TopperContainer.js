import { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import getFilm, { loadGenres } from './actions';
import Topper, { TopperVoid } from './Topper';

class TopperContainer extends Component {
  constructor(props) {
    super(props);
    const { loadLatest, loadList } = this.props;
    loadList();
    loadLatest();
  }

  genGenreString = (genres, filmGenres) => {
    let result = '';
    filmGenres.forEach((elem) => {
      const genre = genres.find((type) => {
        if (type.id === elem) return true;
        return false;
      });
      result = result.concat(genre.name, ' ');
    });
    return result.slice(0, -1);
  }

  render() {
    const { latestFilm, genList } = this.props;
    if (latestFilm.results !== undefined && genList.genres !== undefined) {
      return Topper(latestFilm, genList, this.genGenreString);
    }
    return TopperVoid();
  }
}

TopperContainer.defaultProps = {
  loadLatest: () => {},
  loadList: () => {},
  latestFilm: {},
  genList: {},
};

TopperContainer.propTypes = {
  loadLatest: PropTypes.func,
  loadList: PropTypes.func,
  latestFilm: PropTypes.objectOf(PropTypes.any),
  genList: PropTypes.objectOf(PropTypes.any),
};

const mapStateToProps = (store) => ({
  latestFilm: store.topperReducer.latestFilm,
  genList: store.topperReducer.genList,
});

const mapDispatchToProps = (dispatch) => ({
  loadLatest: () => dispatch(getFilm()),
  loadList: () => dispatch(loadGenres()),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(TopperContainer);
