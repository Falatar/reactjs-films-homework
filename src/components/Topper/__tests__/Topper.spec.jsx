import React from 'react';
import renderer from 'react-test-renderer';
import Topper from '../Topper';

describe('Topper', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Topper />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
