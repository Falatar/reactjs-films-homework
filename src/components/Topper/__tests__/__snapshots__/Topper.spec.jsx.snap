// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="topper"
  id="space"
>
  <div
    className="searchLine"
  >
    <big
      className="Title"
    >
      FILMS
    </big>
    <div
      className="Search"
    >
      <form
        className="SearchForm"
      >
        <input
          className="SearchText"
          placeholder="Search here..."
          type="text"
        />
        <button
          className="SearchButton"
          id="push"
          type="submit"
        >
          âŒ•
        </button>
      </form>
    </div>
  </div>
  <div
    className="filmData"
  >
    <div
      className="Film"
    >
      <big
        className="Name"
      >
        THE JUNGLE BOOK
      </big>
      <h1
        className="Genres"
      >
        Adventure Drama Family Fantasy | 1h 46m
      </h1>
      <div
        className="RatePanel"
      >
        <div
          className="rate"
        >
          <span
            className="star"
          />
          <span
            className="star"
          />
          <span
            className="star"
          />
          <span
            className="star"
          />
          <span
            className="star"
          />
        </div>
        <h1
          className="Rate"
        >
          4.8
        </h1>
      </div>
    </div>
    <div
      className="info"
    >
      <button
        className="Watch"
        id="watch"
        type="button"
      >
        Watch now
      </button>
      <button
        className="View"
        id="check"
        type="button"
      >
        View info
      </button>
    </div>
  </div>
</div>
`;
